tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Summary.java

```diff
@@ -48,57 +64,163 @@
+            return showSnapshot(request.snapshotId);
+        }
+        SummaryResponse sr = getPrItSu(cs, request.loadingPolicy);
+        if (SurveySnapshot.SNAP_CREATE.equals(request.snapshotPolicy)
+            && sr.status == VettingViewerQueue.Status.READY) {
+            saveSnapshot(sr);
+        }
+        return Response.ok(sr, MediaType.APPLICATION_JSON).build();
+    }
+
+    /**
+     * Get the response for Priority Items Summary (new, not an already existing snapshot)
+     *
+     * @param cs the CookieSession identifying the user
+     * @param loadingPolicy the LoadingPolicy
+     * @return the SummaryResponse
+     *
+     * @throws IOException
+     * @throws JSONException
+     */
+    private SummaryResponse getPrItSu(CookieSession cs, LoadingPolicy loadingPolicy) throws IOException, JSONException {
+        VettingViewerQueue.Status status[] = new VettingViewerQueue.Status[1];
+        StringBuilder sb = new StringBuilder();
+        VettingViewerQueue vvq = VettingViewerQueue.getInstance();
+        String str = vvq.getPriorityItemsSummaryOutput(cs, status, loadingPolicy, sb);
+        SummaryResponse sr = new SummaryResponse();
+        sr.status = status[0];
+        sr.ret = str;
+        sr.output = sb.toString();
+        return sr;
+    }
+
+    /**
+     * Return a Response containing the snapshot with the given id
+     *
+     * @param snapshotId
+     * @return the Response
+     */
+    private Response showSnapshot(String snapshotId) {
+        final String json = snap.get(snapshotId);
+        if (json == null) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+        return Response.ok(json).build();
+    }
+
+    /**
+     * Assign a new snapshot id to sr, then save sr as a json snapshot
+     *
+     * @param sr the SummaryResponse
+     */
+    private void saveSnapshot(SummaryResponse sr) {
+        sr.snapshotId = SurveySnapshot.newId();
+        final String json = jsonb.toJson(sr);
+        if (json != null && !json.isEmpty()) {
+            snap.put(sr.snapshotId, json);
+        }
+    }
+
+    @GET
+    @Path("/snapshots")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+        summary = "List All Snapshots",
+        description = "Get a list of all available snapshots of the Priority Items Summary")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "200",
+                description = "Snapshot List",
+                content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = SnapshotListResponse.class)))
+        })
+    public Response listSnapshots(
+        @HeaderParam(Auth.SESSION_HEADER) String sessionString) {
+        CookieSession cs = Auth.getSession(sessionString);
+        if (cs == null) {
+            return Auth.noSessionResponse();
+        }
+        if (!UserRegistry.userIsAdmin(cs.user)) {
```

Comment:
ideally these would be indirected, `userCanReadSnapshots` somewhere/somehow.

Reply:
yes -- this is just a first step and it remains to be decided who will have the right to create snapshots (manually, if anyone) and who will have the right to view them -- presumably two distinct rights...